src/main/kotlin/com/dprint/services/editorservice/EditorServiceManager.kt

```diff
@@ -0,0 +1,78 @@
+package com.dprint.services.editorservice
+
+import com.dprint.core.Bundle
+import com.dprint.core.FileUtils
+import com.dprint.services.NotificationService
+import com.dprint.services.editorservice.v4.EditorServiceV4
+import com.intellij.execution.configurations.GeneralCommandLine
+import com.intellij.execution.util.ExecUtil
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.logger
+import com.intellij.openapi.project.Project
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.int
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+
+private val LOGGER = logger<EditorServiceManager>()
+private const val SCHEMA_V4 = 4
+
+@Service
+class EditorServiceManager(private val project: Project) {
+    private var editorService: EditorService? = null
+    private val notificationService = project.service<NotificationService>()
+
+    init {
+        maybeInitialiseEditorService()
+    }
```

Comment:
TODO: This is throwing "Synchronous execution on EDT" errors, we will need to defer this to the `maybeGetEditorService` method and ensure it is not called on the EDT thread.